Assets/ZSSRichTextEditor.js

```diff
@@ -273,49 +273,66 @@ ZSSEditor.getJoinedCaretArguments = function() {
     return joinedArguments;
 };
 
-ZSSEditor.getYCaretInfo = function() {
-    var y = 0, height = 0;
+ZSSEditor.getCaretYPosition = function() {
     var sel = window.getSelection();
-    if (sel.rangeCount) {
+    // Next line is comented to prevent deselecting selection. It looks like work but if there are any issues will appear then uconmment it as well as code above.
+    //sel.collapseToStart();
+    var range = sel.getRangeAt(0);
+    var span = document.createElement('span');// something happening here preventing selection of elements
+    range.insertNode(span);
+    var topPosition = span.offsetTop;
+    span.parentNode.removeChild(span);
+    return topPosition;
+}
+
+ZSSEditor.getYCaretInfo = function() {
+    var y = 0;
+    var height = 0;
+    var selection = window.getSelection();
+    
+    if (! selection.rangeCount) { // if no selection available just return 0
```

Comment:
Nice check, good addition.

One note though, when comments are used to explain what the code is doing, it usually means that readability can be improved.

For example, instead of:

```javascript

ZSSEditor.getYCaretInfo = function() {
   var y = 0;
   var height = 0;
   var selection = window.getSelection();

   if (! selection.rangeCount) { // if no selection available just return 0
   // ...etc...
}
```

...use...

```javascript
/*
 * @returns {0, 0} if there is no selection.
 */
ZSSEditor.getYCaretInfo = function() {
   var y = 0;
   var height = 0;
   var selection = window.getSelection();

   var noSelection = !selection.rangeCount;

   if (noSelection) {
   // ...etc...
}
```